### DISCUSSION TITLE
One application with several pods with different templates

### DISCUSSION THREAD

**Discussion by shieldlin** [argoproj/argo-rollouts#3594] [General]
I have one application with about 20 pods. Recently, I need to change one of the pods from normal mode to 'monitored' mode by changing the JAVA_TOOL_OPTIONS. This change seems to be permanent. But rollout only have one template for its pods. I find it so complicated to set another application with only one monitored pod. Is there any simple way to perform this action?

**Comment by kostis-codefresh** [comment]
Both Rollout and Deployment are stateless. pods can be killed and restarted at any point by the cluster. There is no guarantee at the order or number of pods.

What is the requirement behind just one pod? What does the option do? 

I feel that this question is about Kubernetes in general, and it is unrelated to Argo Rollouts. Or are you asking to gradually deploy this change to the other pods as well?
